### ISSUE TITLE
[a11y] Links in markdown aren't labeled

### ISSUE THREAD

**Issue by melanie-exygy** [civiform/civiform#9679]
**Describe the bug**
Links that are in any admin-provided markdown don't have labels.  This is a problem for screen reader users because the screen reader just announces that there is a link, but there's no way to know where the link goes.  Markdown can be used in static text questions, program long descriptions and elsewhere.

**To Reproduce**

1. As an admin, add a link in the long description of a program.
2. Publish your changes.
3. If you're in the North Star, go to "/programs/<your-program-name>"
4. Turn the screen reader on.
5. Tab to the link in the long description.
6. Note that the screen reader doesn't say where the link goes.

**Expected behavior**
The screen reader should announce the link text.

**Screenshots**

https://github.com/user-attachments/assets/7035b2c9-b115-4305-9e4d-ae5d3342704e

**Additional context**
Tested with VoiceOver on MacOs.

**Comment by rockycodes** [comment]
We have a method for adding an aria label in `TextFormatter` that might help: https://github.com/civiform/civiform/blob/ebb1e40f691a4bd115abdcd10ae85b179446ac5c/server/app/views/components/TextFormatter.java#L27

**Comment by melanie-exygy** [comment]
This has been fixed.
